src/common/merge.ts

```diff
@@ -177,16 +183,15 @@ export const tryMerge = async (
   } else if (merged) {
     logInfo(`Pull request is already merged.`);
   } else if (
-    /*
-     * TODO(@platform) [2021-08-01] Start pulling the value once it reaches
-     * GA.
-     */
-    mergeStateStatus !== undefined &&
-    mergeStateStatus !== 'CLEAN'
+    requiresStrictStatusChecks === true &&
+    mergeStateStatus === 'BEHIND'
   ) {
+    logInfo(`Pull request branch is behind base branch.`);
+  } else if (mergeStateStatus !== undefined && mergeStateStatus !== 'CLEAN') {
     logInfo(
-      'Pull request cannot be merged cleanly. ' +
-        `Current state: ${mergeStateStatus}.`,
+      `Pull request cannot be merged cleanly. Current state: ${
+        mergeStateStatus as string
+      }.`,
```

Comment:
Do you think it would be a good idea to summarize this into a single check?

Basically, if `mergeStateStatus !== 'CLEAN'`, the merge should fail and print out the value of the field in the logs.

Reply:
It should fail if `requiresStrictStatusChecks` is `true` and `mergeStateStatus` is present and not `CLEAN`

Reply:
https://github.com/ridedott/merge-me-action/pull/946/files#diff-58a9d5fd22661c08d9ae5478f88deeaf642c88f31ada5815bd4dc6785cc8bdeaR186-R193